File: Rust/Easy/CompileError/find-the-middle-index-in-array_6.rs
impl Solution {
    pub fn find_middle_index(nums: Vec<i32>) -> i32 {
        
        let mut left_sum = 0;
        let mut right_sum = nums.iter().sum();
        for (index, num) in nums.iter().enumerate() {
            right_sum -= num;
            if left_sum == right_sum {
                return index as i32;
            }
            left_sum += num;
        }
        -1
    }
}

File: C++/Easy/Accepted/find-the-middle-index-in-array_6.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& nums) {
        
        int n = nums.size();
        if (n == 0) return -1;
        if (n == 1) return 0;
        vector<int> left(n, 0);
        vector<int> right(n, 0);
        left[0] = nums[0];
        right[n - 1] = nums[n - 1];
        for (int i = 1; i < n; i++) {
            left[i] = left[i - 1] + nums[i];
        }
        for (int i = n - 2; i >= 0; i--) {
            right[i] = right[i + 1] + nums[i];
        }
        for (int i = 0; i < n; i++) {
            if (left[i] == right[i]) return i;
        }
        return -1;
    }
};